### PR TITLE
Signature Verification Endpoint

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
In order for the OME to easily verify the signature of an order upon receiving it, and before placing it in to the book, it should be able to query the executioner.

# Changes
- adds a check endpoint to validate signatures
- adds a validate signature function relying on tracer-utils

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -63,6 +64,21 @@ app.post('/submit', async (req, res) => {
...
 
+/**
+ * Endpoint for validation of orders. Used by the OME to validate signatures before 
+ * passing on the orders to the book. This rejects invalid orders early to avoid
+ * false liquidity on the books.
+ */
+app.post('/check', async (req, res) => {
+    if (!req.body.order || !req.body.trader || !req.body.network || !req.body.sig) {
+        console.log("missing params")
+        return res.status(500).send({ error: "Invalid params provided" })
```
Comment (sporejack) [inline index.js:75]: No server error has occurred, this should be a [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400).

File: index.js [hunk 2]
```diff
@@ -63,6 +64,21 @@ app.post('/submit', async (req, res) => {
...
+    }
+    let isValidSig = validateSignature(req.body.order, req.body.trader, req.body.network, req.body.sig)
+    console.log(isValidSig)
+    return res.status(200).send({ verified: isValidSig })
```
Comment (sporejack) [inline index.js:79]: In the event of an _invalid_ signature, the executioner will respond with a HTTP 200 OK when it should likely error.

Comment (sporejack) [reply index.js:79]: Likely a 400 also.

**Review by sporejack — CHANGES_REQUESTED** [review]
API endpoint returns misleading HTTP statuses.

File: index.js (hunk 2 above)
Comment (raymogg) [reply index.js:79]: I'd disagree here @sporejack, isn't 400 reserved for bad request? This is a valid request, the result is just the signature is invalid but the server acted as expected

Comment (sporejack) [reply index.js:79]: @raymogg This is a fair enough call in my opinion (and I did contemplate this before making this comment), but it stills feels awkward having an invalid signature return as a success. The alternative is to just boot it up the protocol abstraction layer (which is the current state of things on this line).

**Review by sporejack — APPROVED** [review]

**Pull request merged**